### ISSUE TITLE
Implement soft argmin layer

### ISSUE THREAD

**Issue by sarambulo** [nikitachaudharicodes/ProjeXion#17]
Create a `soft_argmin.py` module inside the `models` directory that implements the class `SoftArgmin`. This class should inherit from `torch.nn.Module` and should take as input a HxWxD embedding and generate an output embedding of size HxW by first calculating the softmax over the depth dimension (dim=2, zero-indexed) and then calculating the expected value of the depth using those probabilities and the depth values.

![Image](https://github.com/user-attachments/assets/9c025a8a-5ee8-48c4-bd29-af8bd163c2a0)
